libnopegl/src/nodes.c

```diff
@@ -48,6 +48,7 @@ NGLI_STATIC_ASSERT(node_anim_vec_flt,    NGL_NODE_ANIMATEDVEC4     - NGL_NODE_AN
 
 extern const struct param_specs ngli_params_specs[];
 
+/* Warning: there is an assumption that these parameters do not include any nodes */
```

Comment:
nit: something `The common node parameters *must* not include any node-based parameter`

Reply:
Maybe we can put an assert when processing the root node ? (so it will be run once) ?

Reply:
- Comment updated.
- Added a dedicated check function in `gen_specs.c` (see also the added associated commit `gen_specs: remove ret variable shadowing`)